libs/stream/streammanager.cpp

```diff
@@ -265,6 +265,13 @@ bool StreamManager::updateProperties()
  * Binned frame must be sent from the camera driver for this to work consistentaly for all drivers.*/
 void StreamManager::newFrame(const uint8_t * buffer, uint32_t nbytes)
 {
+    // close the data stream on the same thread as the data stream
+    if (m_isRecordingAboutToClose)
+    {
+        stopRecording();
+        return;
```

Comment:
Also, i added the "return" in about to close checking to avoid referencing a buffer that might be destroyed.